pkg/controllers/jobset_controller.go

```diff
@@ -594,64 +595,13 @@ func constructJob(js *jobset.JobSet, rjob *jobset.ReplicatedJob, jobIdx int) (*b
 		job.Spec.Template.Spec.Subdomain = js.Spec.Network.Subdomain
 	}
 
-	// If this job should be exclusive per topology, set the pod affinities/anti-affinities accordingly.
-	if topologyDomain, ok := js.Annotations[jobset.ExclusiveKey]; ok {
-		setExclusiveAffinities(job, topologyDomain)
-	}
 	// if Suspend is set, then we assume all jobs will be suspended also.
 	jobsetSuspended := js.Spec.Suspend != nil && *js.Spec.Suspend
 	job.Spec.Suspend = ptr.To(jobsetSuspended)
 
 	return job, nil
 }
 
-// Appends pod affinity/anti-affinity terms to the job pod template spec,
```

Comment:
I remember a discussion with @vsoch around these annotations.  Should we implement the new way of placements in case of users using these annotations rather than deleting this entirely?

Reply:
I'm not familiar with the context of this PR, but I think a feature of jobset definitely should be to make it easy to ask for common configurations for affinity (or other similar needs for placement). In my case, I did try using one of the defaults here, but it was slightly off for what I needed (so I couldn't use this strategy) and had to implement an entire function akin to the below, with tweaks. I would say it would make sense to:

1. Define a list of common use cases for affinity / anti-affinity
2. Associate them with names the user can ask for in a section of the spec
3. Implement so it's ridiculously easy to just request a named set!

And in this case, it would make sense to only deprecate the existing when an alternative is available. It might be unlikely that someone is using them, in which case maybe OK to delete first, but perhaps adding the new sets first makes sense so if anyone _is_ using them they aren't really angry :)